dix/dispatch.c

```diff
@@ -2703,23 +2706,18 @@ ProcAllocColorCells(ClientPtr client)
 #endif /* XINERAMA */
         {
             xAllocColorCellsReply rep = {
-                .type = X_Reply,
-                .sequenceNumber = client->sequence,
-                .length = bytes_to_int32(length),
                 .nPixels = npixels,
                 .nMasks = nmasks
             };
             if (client->swapped) {
-                swaps(&rep.sequenceNumber);
-                swapl(&rep.length);
                 swaps(&rep.nPixels);
                 swaps(&rep.nMasks);
                 SwapLongs(ppixels, length / 4);
             }
-            WriteToClient(client, sizeof(rep), &rep);
-            WriteToClient(client, length, ppixels);
+
+            X_SEND_REPLY_WITH_RPCBUF(client, rep, rpcbuf);
         }
-        free(ppixels);
+        x_rpcbuf_clear(&rpcbuf);
```

Comment:
Should this only be called if `X_SEND_REPLY_WITH_RPCBUF` wasn't called before? Maybe make it `return X_SEND_REPLY_WITH_RPCBUF(client, rep, rpcbuf)` at the line 2718?

Or we're going to remove that `XINERAMA` stuff first?